spec/rails_helper.rb

```diff
@@ -65,4 +65,22 @@
   config.filter_rails_from_backtrace!
   # arbitrary gems may also be filtered via:
   # config.filter_gems_from_backtrace("gem name")
+
+  def login_user(user)
+    post user_session_path, params: {
+      user: {
+        email: user.email, password: user.password
+      }
+    }
+    follow_redirect!
+  end
+
+  def login_admin(admin)
```

Comment:
Nie wszystko z tego arta polecam, ale moze byc inspiracja https://github.com/heartcombo/devise/wiki/How-To:-Test-controllers-with-Rails-(and-RSpec)